js_modules/dagit/packages/core/src/assets/groupByPartition.tsx

```diff
@@ -34,10 +36,32 @@ export const groupByPartition = (
   return partitionKeys
     .filter((key) => key !== NO_PARTITION_KEY)
     .map((key) => {
-      const [latest, ...predecessors] = [...(grouped[key] || [])].sort(sortByEventTimestamp);
+      const sorted = [...(grouped[key] || [])].sort(sortByEventTimestamp);
+      const latestMaterialization = sorted.find((a) => a.__typename === 'MaterializationEvent');
+      let latest = latestMaterialization || sorted[0] || null;
+
+      if (latest && latest.__typename === 'MaterializationEvent') {
+        const observationsAboutLatest = sorted.filter(
+          (e) =>
+            e.__typename === 'ObservationEvent' && Number(e.timestamp) > Number(latest.timestamp),
+        );
+
+        latest = {...latest, metadataEntries: [...latest.metadataEntries]};
+        for (const observation of observationsAboutLatest) {
+          latest.metadataEntries.push(
+            ...observation.metadataEntries.map((e) => ({
+              ...e,
```

Comment:
I think it would be good to make it clear that the observations were yielded after the materialization in the "grouped" view. Can we add a timestamp of the observation, or a link to the run it was yielded from?